src/utils/ChatUtils.ts

```diff
@@ -21,6 +23,31 @@ export function serialize_chat(chat: ChatItem[], activeChars: ChatChar[]): strin
   }, undefined, 2);
 }
 
+export async function serialize_chat_with_custom_chars(chat: ChatItem[], activeChars: ChatChar[], customChars: Character[]): Promise<string> {
```

Comment:
1. Don't define a new function, just modify `serialize_chat`
2. Why is this function `async`?